### ISSUE TITLE
'namespace syslog { }' redeclared as a different kind of symbol

### ISSUE THREAD

**Issue by grepsuzette** [snowkit/linc#14]
Hello, I just discovered linc and it's awesome.
For my first contribution I wanted to create a **linc_syslog** lib. 
But I've got some problem to compile it, namespace stuffs. I have a hard time figuring out what's wrong, since the hx files use package syslog and the cpp use linc::syslog, the relationship is not very clear for me from the linc_empty template. Here are the full errors with `haxe build.hxml`:

```
Compiling group: haxe
g++ -Iinclude -I/home/mike/code/haxe/linc_syslog//linc/ -DSYSLOG_EXAMPLE_DEFINE_FOR_THE_CPP_CODE -c -fvisibility=hidden -O2 -fpic -fPIC -Wno-overflow -DHX_LINUX -DHXCPP_M64 -DHXCPP_VISIT_ALLOCS(haxe) -DHXCPP_API_LEVEL=331(haxe) -m64 -DHXCPP_M64 -I/usr/share/haxelib/hxcpp/4,0,4/include ... tags=[haxe,static]
- src/Test.cpp 
Error: In file included from /usr/share/haxelib/hxcpp/4,0,4/include/hxcpp.h:332:
include/syslog/Option.h:9:19: error: ‘namespace syslog { }’ redeclared as different kind of symbol
HX_DECLARE_CLASS1(syslog,Option)
^~~~~~
/usr/share/haxelib/hxcpp/4,0,4/include/hx/Macros.h:18:48: note: in definition of macro ‘HX_DECLARE_CLASS1’
#define HX_DECLARE_CLASS1(ns1,klass) namespace ns1 { HX_DECLARE_CLASS0(klass) }
^~~
In file included from /usr/include/syslog.h:1,
from /home/mike/code/haxe/linc_syslog//linc/linc_syslog.h:10,
from ./src/Test.cpp:6:
/usr/include/sys/syslog.h:190:13: note: previous declaration ‘void syslog(int, const char*, ...)’
extern void syslog (int __pri, const char *__fmt, ...)
^~~~~~
In file included from ./src/Test.cpp:15:
include/syslog/Option.h:10:11: error: ‘namespace syslog { }’ redeclared as different kind of symbol
namespace syslog{
^~~~~~
In file included from /usr/include/syslog.h:1,
from /home/mike/code/haxe/linc_syslog//linc/linc_syslog.h:10,
from ./src/Test.cpp:6:
/usr/include/sys/syslog.h:190:13: note: previous declaration ‘void syslog(int, const char*, ...)’
extern void syslog (int __pri, const char *__fmt, ...)
^~~~~~
In file included from ./src/Test.cpp:15:
include/syslog/Option.h:27:12: error: ‘syslog’ in namespace ‘::’ does not name a type
static ::syslog::Option LogCons;
^~~~~~
In file included from /usr/include/syslog.h:1,
from /home/mike/code/haxe/linc_syslog//linc/linc_syslog.h:10,
from ./src/Test.cpp:6:
/usr/include/sys/syslog.h:190:13: note: ‘void syslog(int, const char*, ...)’ declared here
extern void syslog (int __pri, const char *__fmt, ...)
^~~~~~
In file included from ./src/Test.cpp:15:
include/syslog/Option.h:28:19: error: ‘syslog’ in namespace ‘::’ does not name a type
static inline ::syslog::Option LogCons_dyn() { return LogCons; }
^~~~~~
In file included from /usr/include/syslog.h:1,
from /home/mike/code/haxe/linc_syslog//linc/linc_syslog.h:10,
from ./src/Test.cpp:6:
/usr/include/sys/syslog.h:190:13: note: ‘void syslog(int, const char*, ...)’ declared here
extern void syslog (int __pri, const char *__fmt, ...)
^~~~~~
In file included from ./src/Test.cpp:15:
include/syslog/Option.h:29:12: error: ‘syslog’ in namespace ‘::’ does not name a type
static ::syslog::Option LogNDelay;
^~~~~~
In file included from /usr/include/syslog.h:1,
from /home/mike/code/haxe/linc_syslog//linc/linc_syslog.h:10,
from ./src/Test.cpp:6:
/usr/include/sys/syslog.h:190:13: note: ‘void syslog(int, const char*, ...)’ declared here
extern void syslog (int __pri, const char *__fmt, ...)
^~~~~~
In file included from ./src/Test.cpp:15:
include/syslog/Option.h:30:19: error: ‘syslog’ in namespace ‘::’ does not name a type
static inline ::syslog::Option LogNDelay_dyn() { return LogNDelay; }
^~~~~~
In file included from /usr/include/syslog.h:1,
from /home/mike/code/haxe/linc_syslog//linc/linc_syslog.h:10,
from ./src/Test.cpp:6:
/usr/include/sys/syslog.h:190:13: note: ‘void syslog(int, const char*, ...)’ declared here
extern void syslog (int __pri, const char *__fmt, ...)
^~~~~~
In file included from ./src/Test.cpp:15:
include/syslog/Option.h:31:12: error: ‘syslog’ in namespace ‘::’ does not name a type
static ::syslog::Option LogODelay;
^~~~~~
In file included from /usr/include/syslog.h:1,
from /home/mike/code/haxe/linc_syslog//linc/linc_syslog.h:10,
from ./src/Test.cpp:6:
/usr/include/sys/syslog.h:190:13: note: ‘void syslog(int, const char*, ...)’ declared here
extern void syslog (int __pri, const char *__fmt, ...)
^~~~~~
In file included from ./src/Test.cpp:15:
include/syslog/Option.h:32:19: error: ‘syslog’ in namespace ‘::’ does not name a type
static inline ::syslog::Option LogODelay_dyn() { return LogODelay; }
^~~~~~
In file included from /usr/include/syslog.h:1,
from /home/mike/code/haxe/linc_syslog//linc/linc_syslog.h:10,
from ./src/Test.cpp:6:
/usr/include/sys/syslog.h:190:13: note: ‘void syslog(int, const char*, ...)’ declared here
extern void syslog (int __pri, const char *__fmt, ...)
^~~~~~
In file included from ./src/Test.cpp:15:
include/syslog/Option.h:33:12: error: ‘syslog’ in namespace ‘::’ does not name a type
static ::syslog::Option LogPError;
^~~~~~
In file included from /usr/include/syslog.h:1,
from /home/mike/code/haxe/linc_syslog//linc/linc_syslog.h:10,
from ./src/Test.cpp:6:
/usr/include/sys/syslog.h:190:13: note: ‘void syslog(int, const char*, ...)’ declared here
extern void syslog (int __pri, const char *__fmt, ...)
^~~~~~
In file included from ./src/Test.cpp:15:
include/syslog/Option.h:34:19: error: ‘syslog’ in namespace ‘::’ does not name a type
static inline ::syslog::Option LogPError_dyn() { return LogPError; }
^~~~~~
In file included from /usr/include/syslog.h:1,
from /home/mike/code/haxe/linc_syslog//linc/linc_syslog.h:10,
from ./src/Test.cpp:6:
/usr/include/sys/syslog.h:190:13: note: ‘void syslog(int, const char*, ...)’ declared here
extern void syslog (int __pri, const char *__fmt, ...)
^~~~~~
In file included from ./src/Test.cpp:15:
include/syslog/Option.h:35:12: error: ‘syslog’ in namespace ‘::’ does not name a type
static ::syslog::Option LogPid;
^~~~~~
In file included from /usr/include/syslog.h:1,
from /home/mike/code/haxe/linc_syslog//linc/linc_syslog.h:10,
from ./src/Test.cpp:6:
/usr/include/sys/syslog.h:190:13: note: ‘void syslog(int, const char*, ...)’ declared here
extern void syslog (int __pri, const char *__fmt, ...)
^~~~~~
In file included from ./src/Test.cpp:15:
include/syslog/Option.h:36:19: error: ‘syslog’ in namespace ‘::’ does not name a type
static inline ::syslog::Option LogPid_dyn() { return LogPid; }
^~~~~~
In file included from /usr/include/syslog.h:1,
from /home/mike/code/haxe/linc_syslog//linc/linc_syslog.h:10,
from ./src/Test.cpp:6:
/usr/include/sys/syslog.h:190:13: note: ‘void syslog(int, const char*, ...)’ declared here
extern void syslog (int __pri, const char *__fmt, ...)
^~~~~~
./src/Test.cpp: In static member function ‘static void Test_obj::main()’:
./src/Test.cpp:39:17: error: ‘::syslog’ is not a class, namespace, or enumeration
HXLINE(   8)  ::syslog::Syslog_obj::open(HX_("syslog-test",08,b2,69,17),::Array_obj< ::Dynamic>::__new(2)->init(0,::syslog::Option_obj::LogPid_dyn())->init(1,::syslog::Option_obj::LogCons_dyn()),(int)8);
^~~~~~
./src/Test.cpp:39:117: error: ‘::syslog’ is not a class, namespace, or enumeration
HXLINE(   8)  ::syslog::Syslog_obj::open(HX_("syslog-test",08,b2,69,17),::Array_obj< ::Dynamic>::__new(2)->init(0,::syslog::Option_obj::LogPid_dyn())->init(1,::syslog::Option_obj::LogCons_dyn()),(int)8);
^~~~~~
./src/Test.cpp:39:161: error: ‘::syslog’ is not a class, namespace, or enumeration
HXLINE(   8)  ::syslog::Syslog_obj::open(HX_("syslog-test",08,b2,69,17),::Array_obj< ::Dynamic>::__new(2)->init(0,::syslog::Option_obj::LogPid_dyn())->init(1,::syslog::Option_obj::LogCons_dyn()),(int)8);
^~~~~~
./src/Test.cpp:40:95: error: too many arguments to function ‘void linc::syslog::_syslog(int, const char*)’
HXLINE(   9)  linc::syslog::_syslog((int)4,HX_("Hello this is just a test",03,72,0a,54),null());
^
In file included from ./src/Test.cpp:6:
/home/mike/code/haxe/linc_syslog//linc/linc_syslog.h:18:14: note: declared here
void _syslog(int pri, const char *s);
^~~~~~~
Error: Build failed
```
Since all of this is probably meaningless without the context, I've pushed my project to https://github.com/emugel/linc_syslog
The test compiles provided I don't use classes so I may have missed something.

Also, is there any gitter for linc?
Looking forward to contributing!

**Comment by ruby0x1** [comment]
Hi,
There's a few errors above it seems.
The first is that you have two things named the same, in the same level/scope.
This is not allowed:
```cpp
int name;
void name() {}
```
The symbol (`name`, a variable) is trying to be defined a second time, but this time as something else (a function). The same is true for the namespace. You already have a symbol (`syslog`) in scope that is named that, you can't create another symbol name `syslog`. 

That means you need a different name for the namespace!

The other errors seem like to many arguments to a function, but it's easier to fix the main ones and see what stays behind.

**Comment by grepsuzette** [comment]
It now compiles (and works, at least basically: I can see the log entry in my Archlinux journalctl). Thank you!

I had a few comments and questions before closing this thread.

1) I am a little annoyed by the fact I had to use a namespace name such as 'haxe_syslog'. I picked the name out of the blue, but I wondered if you knew of more established convention to name those? I've looked into your `linc_sdl` but seems you didn't have a problem to use the `sdl` namespace, I don't know about your other projects.

2) Is there a gitter for linc? (I found a snowkit one).

3) When this is tested and finalized, can I submit this even though I only have Linux? I think I have seen you also want to support Mac and Windows.

4) I think a few months ago I saw a linc to play music modules (.mod, .xm) a la modplug. But it's not in the list anymore. Or was I confusing it with the midi linc?

That's all I can think about :)
Thanks again.
